tembo-pgmq-python/tembo_pgmq_python/queue.py

```diff
@@ -25,6 +37,36 @@ class QueueMetrics:
     scrape_time: datetime
 
 
+def transaction(func: Callable) -> Callable:
+    """Decorator to run a method within a database transaction."""
+
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        perform_transaction = kwargs.pop("perform_transaction", self.perform_transaction)
+        if perform_transaction:
+            with self.pool.connection() as conn:
+                try:
+                    logger.debug(f"Transaction started with conn: {conn}")
+                    with conn.transaction():
+                        result = func(self, *args, conn=conn, **kwargs)
+                    logger.debug(f"Transaction completed with conn: {conn}")
+                    return result
```

Comment:
I don't fully grasp this code, too. What makes it not start a transaction every single call?